boggs/src/main/java/com/hartwig/healthchecks/boggs/healthcheck/mapping/MappingExtractor.java

```diff
@@ -17,65 +8,73 @@
 import com.hartwig.healthchecks.common.util.CheckType;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
 
 public class MappingExtractor extends BoggsExtractor {
-	private static final String EMPTY_FILES_ERROR = "Found empty Summary files and/or fastqc_data under path -> %s";
+    private static final String EMPTY_FILES_ERROR = "Found empty Summary files and/or fastqc_data under path -> %s";
 
-	private static final String REALIGN = "realign";
-	private static final String SAMPLE_PREFIX = "CPCT";
-	private static final String REF_SAMPLE_SUFFIX = "R";
-	private static final String FLAGSTAT_SUFFIX = ".flagstat";
+    private static final String REALIGN = "realign";
+    private static final String SAMPLE_PREFIX = "CPCT";
+    private static final String REF_SAMPLE_SUFFIX = "R";
+    private static final String FLAGSTAT_SUFFIX = ".flagstat";
 
-	@NotNull
-	private final FlagStatParser flagstatParser;
+    @NotNull
+    private final FlagStatParser flagstatParser;
 
-	public MappingExtractor(@NotNull FlagStatParser flagstatParser) {
-		this.flagstatParser = flagstatParser;
-	}
+    public  MappingExtractor(@NotNull final FlagStatParser flagstatParser) {
+        this.flagstatParser = flagstatParser;
+    }
 
-	public MappingReport extractFromRunDirectory(@NotNull final String runDirectory)
-			throws IOException, EmptyFileException {
+    public MappingReport extractFromRunDirectory(@NotNull final String runDirectory)
+            throws IOException, EmptyFileException {
 
-		Optional<Path> sampleFile = Files.walk(new File(runDirectory).toPath())
-				.filter(p -> p.getFileName().toString().startsWith(SAMPLE_PREFIX)
-						&& p.getFileName().toString().endsWith(REF_SAMPLE_SUFFIX))
-				.findFirst();
-		if (!sampleFile.isPresent()) {
-			throw new FileNotFoundException();
-		}
-		String externalId = sampleFile.get().getFileName().toString();
-		Long totalSequences = sumOfTotalSequences(runDirectory);
-		MappingDataReport mappingDataReport = getFlagstatsData(sampleFile.get(), totalSequences.toString());
-		return new MappingReport(CheckType.MAPPING, externalId, totalSequences.toString(), mappingDataReport);
-	}
+        final Optional<Path> sampleFile = Files.walk(new File(runDirectory).toPath())
+                .filter(p -> p.getFileName().toString().startsWith(SAMPLE_PREFIX)
+                        && p.getFileName().toString().endsWith(REF_SAMPLE_SUFFIX))
+                .findFirst();
+        if (!sampleFile.isPresent()) {
+            throw new FileNotFoundException();
+        }
+        final String externalId = sampleFile.get().getFileName().toString();
+        final Long totalSequences = sumOfTotalSequences(runDirectory);
+        final MappingDataReport mappingDataReport = getFlagstatsData(sampleFile.get(), totalSequences.toString());
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Potential violation of Law of Demeter (object not created locally)](https://www.codacy.com/app/hmf/health-checks/file/2780941525/issues/source?bid=3363458&fileBranchId=3365457#l47)